tap_shopify/streams/base.py

```diff
@@ -204,12 +206,16 @@ def get_objects(self):
                     objects = self.call_api(query_params)
 
                 for obj in objects:
+                    dict_obj = obj.to_dict()
+                    replication_value = strptime_to_utc(dict_obj[self.replication_key])
```

Comment:
If performance is a concern here, `to_dict` may not be the fastest option.

As you can see on the line below this, we can access fields on the object with the dot operator.
```python
    if obj.id < since_id:
```

The Python data model actually gives us a way to dynamically do this via `getattr()`
> Return the value of the named attribute of object. name must be a string. ... getattr(x, 'foobar') is equivalent to x.foobar

Source: [Docs Link](https://docs.python.org/3/library/functions.html#getattr)

The code would then look like
```suggestion
                    replication_value = strptime_to_utc(getattr(obj, self.replication_key))
```
Someone should test the performance a little more thoroughly, but here's what I did to performance test and the results:

```python
>>> from timeit import default_timer as timer
>>> def foo():
>>>   start = timer()
>>>   obj.to_dict()[self.replication_key]
>>>   end = timer()
>>>   return end - start
>>> 
>>> sum([foo() for i in range(100)]) / 100
3.0518800485879184e-06
>>> sum([foo() for i in range(100)]) / 100
3.034400288015604e-06
>>> sum([foo() for i in range(100)]) / 100
3.2068253494799135e-06
>>> 
>>> 
>>> sum([bar() for i in range(100)]) / 100
6.84354454278946e-07
>>> sum([bar() for i in range(100)]) / 100
6.862124428153039e-07
>>> sum([bar() for i in range(100)]) / 100
6.863870657980443e-07
```

Summary
```org
   | trial |                to_dict |               getattr |
   |-------+------------------------+-----------------------|
   |     1 | 3.0518800485879184e-06 |  6.84354454278946e-07 |
   |     2 |  3.034400288015604e-06 | 6.862124428153039e-07 |
   |     3 | 3.2068253494799135e-06 | 6.863870657980443e-07 |
```

Reply:
Used getattr() instead of to_dict()